### PR TITLE
[kvutils] - Extract command deduplication steps from the TransactionCommitter [KVL-1174]

### PR DESCRIPTION
**Pull request by nicu-da**
CHANGELOG_BEGIN

CHANGELOG_END

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [x] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description
- [ ] If you mean to change the status of a component, please make sure you keep [the Component Status page](https://github.com/digital-asset/daml/blob/main/docs/source/support/component-statuses.rst) up to date.

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ledger/participant-state/kvutils/src/test/suite/scala/com/daml/ledger/participant/state/kvutils/committer/transaction/TransactionCommitterSpec.scala [hunk 1]
```diff
@@ -173,149 +164,6 @@ class TransactionCommitterSpec
     }
   }
 
-  "deduplicateCommand" should {
```
Comment (fabiotudone-da) [inline ledger/participant-state/kvutils/src/test/suite/scala/com/daml/ledger/participant/state/kvutils/committer/transaction/TransactionCommitterSpec.scala:176]: These tests seem to have disappeared, missing file?

Comment (nicu-da) [reply ledger/participant-state/kvutils/src/test/suite/scala/com/daml/ledger/participant/state/kvutils/committer/transaction/TransactionCommitterSpec.scala:176]: Good catch, thank you. Added the file

File: ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committer/transaction/CommandDeduplication.scala [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+          DamlStateValue.newBuilder
+            .setCommandDedup(
+              commandDedupBuilder.build
+            )
+            .build,
+        )
+        StepContinue(transactionEntry)
+      }
+    }
+  // Checks that the submission time of the command is after the
```
Comment (fabiotudone-da) [inline ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committer/transaction/CommandDeduplication.scala:102]:
```suggestion

  // Checks that the submission time of the command is after the
```

**Review by fabiotudone-da — APPROVED** [review]

**Review by hubert-da — APPROVED** [review]
Approving with both hands

**Pull request merged**